django/db/models/fields/__init__.py

```diff
@@ -762,17 +766,19 @@ def has_default(self):
         """
         return self.default is not NOT_PROVIDED
 
+    @cached_property
```

Comment:
Ok this won't work with subclasses that override `get_default`, because of the way `cached_property` works. However the spirit can be done with some inner caching logic on an attribute like `self._default`